### PR TITLE
Batch updates and huber loss def

### LINKED ISSUE #18: Speeding the replay
First, thank you for this wonderful code.

In the `replay` function, there is one `model.fit(state, target_f)` per sample in the minibach (i.e. if there are 32 samples, then there are 32 `fit` ).

I think all samples of the minibatch could be used in a single update with one single `train_on_batch(states, targets_f)`, which would speed up the processing time.

### PR DESCRIPTION
**Pull request by pskrunner14**
Implemented training on batches in `dqn.py` as it results in less training time. Fixes #18. Although degrades performance in `ddqn.py` not sure why. 

Fixed Huber Loss def in `ddqn.py` as described in TensorFlow docs:
https://www.tensorflow.org/api_docs/python/tf/losses/huber_loss

### CODE REVIEW

**Comment by pskrunner14** [comment]
@keon I'd appreciate any feedback on the PR. Thanks :)

**Comment by keon** [comment]
Sorry for the late review. @pskrunner14 
I am a little hesitant to merge this PR because I wanted to keep the example at its simplest form, not the most efficient form - for the efficient implementation ppl can take a look at the OpenAI's baseline.
Could you add one more file called dqn_batch.py for this?

**Comment by pskrunner14** [comment]
@keon added dqn_batch.py
